Gradle deploy throws exception in Windows 10

I was trying out the [template-project-gradle](https://github.com/ev3dev-lang-java/template-project-gradle) project on Windows 10. When I executed the `deploy` task, an exception was thrown, saying that "The process cannot access the file because it is being used by another process." The stack trace is attached below.

However, the program seemed to be deployed successfully since it could be run with the `run` task. The problem was not being able to use the `deployRun` task.

A workaround would be executing `deploy` and ignore the error message, and then `run`, which would take one more step.

```
org.gradle.api.tasks.TaskExecutionException: Execution failed for task ':deploy'.
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeActions(ExecuteActionsTaskE
xecuter.java:103)
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.execute(ExecuteActionsTaskExecuter
.java:73)
        at org.gradle.api.internal.tasks.execution.OutputDirectoryCreatingTaskExecuter.execute(OutputDirectoryCr
eatingTaskExecuter.java:51)
        at org.gradle.api.internal.tasks.execution.SkipUpToDateTaskExecuter.execute(SkipUpToDateTaskExecuter.jav
a:59)
        at org.gradle.api.internal.tasks.execution.ResolveTaskOutputCachingStateExecuter.execute(ResolveTaskOutp
utCachingStateExecuter.java:54)
        at org.gradle.api.internal.tasks.execution.ValidatingTaskExecuter.execute(ValidatingTaskExecuter.java:59
)
        at org.gradle.api.internal.tasks.execution.SkipEmptySourceFilesTaskExecuter.execute(SkipEmptySourceFiles
TaskExecuter.java:101)
        at org.gradle.api.internal.tasks.execution.FinalizeInputFilePropertiesTaskExecuter.execute(FinalizeInput
FilePropertiesTaskExecuter.java:44)
        at org.gradle.api.internal.tasks.execution.CleanupStaleOutputsExecuter.execute(CleanupStaleOutputsExecut
er.java:91)
        at org.gradle.api.internal.tasks.execution.ResolveTaskArtifactStateTaskExecuter.execute(ResolveTaskArtif
actStateTaskExecuter.java:62)
        at org.gradle.api.internal.tasks.execution.SkipTaskWithNoActionsExecuter.execute(SkipTaskWithNoActionsEx
ecuter.java:59)
        at org.gradle.api.internal.tasks.execution.SkipOnlyIfTaskExecuter.execute(SkipOnlyIfTaskExecuter.java:54
)
        at org.gradle.api.internal.tasks.execution.ExecuteAtMostOnceTaskExecuter.execute(ExecuteAtMostOnceTaskEx
ecuter.java:43)
        at org.gradle.api.internal.tasks.execution.CatchExceptionTaskExecuter.execute(CatchExceptionTaskExecuter
.java:34)
        at org.gradle.execution.taskgraph.DefaultTaskGraphExecuter$EventFiringTaskWorker$1.run(DefaultTaskGraphE
xecuter.java:256)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(Def
aultBuildOperationExecutor.java:317)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(Def
aultBuildOperationExecutor.java:309)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor.execute(DefaultBuildOperationExecutor.ja
va:185)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor.run(DefaultBuildOperationExecutor.java:9
7)
        at org.gradle.internal.operations.DelegatingBuildOperationExecutor.run(DelegatingBuildOperationExecutor.
java:31)
        at org.gradle.execution.taskgraph.DefaultTaskGraphExecuter$EventFiringTaskWorker.execute(DefaultTaskGrap
hExecuter.java:249)
        at org.gradle.execution.taskgraph.DefaultTaskGraphExecuter$EventFiringTaskWorker.execute(DefaultTaskGrap
hExecuter.java:238)
        at org.gradle.execution.taskgraph.DefaultTaskPlanExecutor$TaskExecutorWorker$1.execute(DefaultTaskPlanEx
ecutor.java:104)
        at org.gradle.execution.taskgraph.DefaultTaskPlanExecutor$TaskExecutorWorker$1.execute(DefaultTaskPlanEx
ecutor.java:98)
        at org.gradle.execution.taskgraph.DefaultTaskExecutionPlan.execute(DefaultTaskExecutionPlan.java:663)
        at org.gradle.execution.taskgraph.DefaultTaskExecutionPlan.executeWithTask(DefaultTaskExecutionPlan.java
:596)
        at org.gradle.execution.taskgraph.DefaultTaskPlanExecutor$TaskExecutorWorker.run(DefaultTaskPlanExecutor
.java:98)
        at org.gradle.internal.concurrent.ExecutorPolicy$CatchAndRecordFailures.onExecute(ExecutorPolicy.java:63
)
        at org.gradle.internal.concurrent.ManagedExecutorImpl$1.run(ManagedExecutorImpl.java:46)
        at org.gradle.internal.concurrent.ThreadFactoryImpl$ManagedThreadRunnable.run(ThreadFactoryImpl.java:55)
Caused by: org.gradle.api.GradleException: Upload of program files failed.
        at com.github.elj.gradle.GradlePlugin$_registerDeploymentTasks_closure5$_closure15$_closure20$_closure21
.doCall(GradlePlugin.groovy:227)
        at com.github.elj.gradle.GradlePlugin$_registerDeploymentTasks_closure5$_closure15$_closure20.doCall(Gra
dlePlugin.groovy:193)
        at org.gradle.api.internal.AbstractTask$ClosureTaskAction.execute(AbstractTask.java:732)
        at org.gradle.api.internal.AbstractTask$ClosureTaskAction.execute(AbstractTask.java:705)
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter$1.run(ExecuteActionsTaskExecuter.j
ava:124)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(Def
aultBuildOperationExecutor.java:317)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(Def
aultBuildOperationExecutor.java:309)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor.execute(DefaultBuildOperationExecutor.ja
va:185)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor.run(DefaultBuildOperationExecutor.java:9
7)
        at org.gradle.internal.operations.DelegatingBuildOperationExecutor.run(DelegatingBuildOperationExecutor.
java:31)
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeAction(ExecuteActionsTaskEx
ecuter.java:113)
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeActions(ExecuteActionsTaskE
xecuter.java:95)
        ... 29 more
Caused by: java.nio.file.FileSystemException: C:\Users\username\AppData\Local\Temp\splash-4198667171730280429.tx
t: The process cannot access the file because it is being used by another process.

        at java_nio_file_Files$delete.call(Unknown Source)
        at com.github.elj.gradle.GradlePlugin$_registerDeploymentTasks_closure5$_closure15$_closure20$_closure21
.doCall(GradlePlugin.groovy:218)
        ... 40 more


* Get more help at https://help.gradle.org

BUILD FAILED in 4s
6 actionable tasks: 6 executed
```

--- a/build.gradle
+++ b/build.gradle
@@ -1,6 +1,7 @@
 plugins {
     id 'java'
     id 'java-gradle-plugin'
+    id 'groovy'
     id 'maven-publish'
 }
 

--- a/build.gradle
+++ b/build.gradle
@@ -5,7 +5,7 @@ plugins {
 }
 
 group 'com.github.ev3dev-lang-java'
-version '1.6.7'
+version '1.6.8'
 
 sourceCompatibility = 1.8
 

--- a/src/main/groovy/com/github/elj/gradle/GradlePlugin.groovy
+++ b/src/main/groovy/com/github/elj/gradle/GradlePlugin.groovy
@@ -7,7 +7,6 @@ import org.gradle.api.Project
 import org.gradle.api.plugins.JavaPlugin
 
 import java.nio.charset.StandardCharsets
-import java.nio.file.StandardCopyOption
 import java.nio.file.Files
 import java.nio.file.Path
 import java.nio.file.Paths
@@ -206,16 +205,10 @@ class GradlePlugin implements Plugin<Project> {
                                 // provide a default splash
                                 Path splash = ext.localSplashPath()
                                 if (!Files.exists(splash)) {
-                                    System.out.println "[Using default splash, copying to a temporary file.]"
-                                    splash = Files.createTempFile("splash-", ".txt");
-
                                     InputStream str = null
-
                                     try {
                                         str = GradlePlugin.class.getResourceAsStream("/splash.txt")
-                                        Files.copy(str, splash, StandardCopyOption.REPLACE_EXISTING)
-                                        put splash, ext.brickSplashPath(), 0644
-                                        Files.delete(splash)
+                                        putStream str, "[default splash]", ext.brickSplashPath(), 0644
                                     } finally {
                                         str?.close()
                                     }

--- a/src/main/groovy/com/github/elj/gradle/SFTP.groovy
+++ b/src/main/groovy/com/github/elj/gradle/SFTP.groovy
@@ -29,6 +29,15 @@ class SFTP implements AutoCloseable {
         }
     }
 
+    void putStream(InputStream stream, String name, String destination, int mode) throws Exception {
+        System.out.println("Uploading file from stream: " + name);
+
+        sftp.with {
+            put stream, destination, OVERWRITE
+            chmod mode, destination
+        }
+    }
+
     void mkdir(String destination) throws Exception {
         try {
             sftp.with {

